### PR TITLE
Disable UsernamePassword Login button when fields are empty

### PR DESCRIPTION
**Pull request by eri9000**
Related to issue: #

### Description:
- Improvements to the Username/Login prompt 

### Summary of changes:
- Disables the "Login" button when the text is empty
- Disallows submitting empty username/password with keyboard action button 
- Adds test to verify behavior 
 
### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [ ] link: https://runtime-kotlin.esri.com/view/vTest/job/vtest/job/toolkit/53
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

**Review by XuemingRocks — COMMENTED** [review]
Overall looks good. However, I can still repro with the following workflow:

1. Input username
2. Tap on password field. The Send button at the bottom right corner of the keyboard is enabled.
3. Tap on the Send button.
image

**Comment by eri9000** [comment]
@XuemingRocks Good catch, I have pushed a fix for that. Have another look.

**Review by XuemingRocks — APPROVED** [review]
LGTM!

File: toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt [hunk 1]
```diff
@@ -113,11 +113,13 @@ public fun UsernamePasswordAuthenticator(
...
-                    usernameFieldText,
-                    passwordFieldText
-                )
-                passwordFieldText = ""
+                if (usernameFieldText.isNotEmpty() && passwordFieldText.isNotEmpty()) {
+                    usernamePasswordChallenge.continueWithCredentials(
+                        usernameFieldText,
+                        passwordFieldText
+                    )
+                    passwordFieldText = ""
```
Comment (gunt0001) [inline toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt:121]: Why is the password text being reset here? Could add a comment.

File: toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt [hunk 2]
```diff
@@ -136,7 +138,6 @@ public fun UsernamePasswordAuthenticator(
                     keyboardType = KeyboardType.Email,
                     imeAction = ImeAction.Next
                 ),
-                keyboardActions = keyboardActions,
```
Comment (gunt0001) [inline toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt:139]: Why this change?

File: toolkit/authentication/src/androidTest/java/com/arcgismaps/toolkit/authentication/UsernamePasswordTests.kt [hunk 3]
```diff
@@ -84,6 +86,31 @@ class UsernamePasswordTests {
         assert(response is NetworkAuthenticationChallengeResponse.ContinueWithCredential)
     }
 
+    /**
+     * Given a Dialog Authenticator
```
Comment (gunt0001) [inline toolkit/authentication/src/androidTest/java/com/arcgismaps/toolkit/authentication/UsernamePasswordTests.kt:90]: Maybe better:
```suggestion
     * Given a UserNamePasswordAuthenticator
```

File: toolkit/authentication/src/androidTest/java/com/arcgismaps/toolkit/authentication/UsernamePasswordTests.kt [hunk 4]
```diff
@@ -84,6 +86,31 @@ class UsernamePasswordTests {
...
+     * When the username or password fields are empty
+     * Then the login button should be disabled
+     * And when the username and password fields are both filled
+     * Then the login button should be enabled
+     *
+     * @since 200.5.0
+     */
+    @Test
+    fun loginButtonEnabledState() = runTest {
+        val response = testUsernamePasswordChallengeWithStateRestoration {
```
Comment (hud10837) [inline toolkit/authentication/src/androidTest/java/com/arcgismaps/toolkit/authentication/UsernamePasswordTests.kt:100]: Does this need to test the full workflow? I think the test would be faster and more focused by just using `composeTestRule.setContent { UsernamePasswordAuthenticator(UsernamePasswordChallenge(...), modifier = Modifier.testTag("testUsernamePasswordAuthenticator")) } and performing the below verifications direclty there rather than actually logging in.

File: toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt (hunk 1 above)
Comment (hud10837) [reply toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt:121]: This was already being done, to erase the password when submitting as many forms do so if the credentials were wrong then you will have to re-enter the password. I suppose this is a design consideration that we never fully discussed.

File: toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt (hunk 2 above)
Comment (hud10837) [reply toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt:139]: I am assuming it's so that you can no longer press send on the keyboard in order to submit the username and password, but I would assume that this case would be handled by the check added to `submitUsernamePassword`, I'm not sure why Xueming was seeing the submission work out. If necessary I think better to add an additional check to the `onSend` in the keyboard actions.

**Review by hud10837 — COMMENTED** [review]
Good change, just a few thoughts.

File: toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt (hunk 2 above)
Comment (eri9000) [reply toolkit/authentication/src/main/java/com/arcgismaps/toolkit/authentication/UsernamePasswordAuthenticator.kt:139]: This was discussed and we agreed it was ok to remove since this `OutlinedTextField` sets the keyboard `imeAction` to `Next` and the `keyboardActions` object modifies `onSend` so it does not directly utilize this anyway.

**Review by gunt0001 — APPROVED** [review]
@eri9000 - great test coverage enhancements 👍

**Review by hud10837 — APPROVED** [review]
LGTM 👍
